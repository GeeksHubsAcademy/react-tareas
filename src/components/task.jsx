import React from 'react';
import './task.scss';
import {connect} from 'react-redux'

export class Task extends React.Component {
  state = {
    newText: this.props.data.text,
  };

  handleChange = ev => {
    this.setState({ newText: ev.target.value });
  };
  isEnter = ev => {
    if (ev.keyCode === 13) {
      this.edit(ev);
    }
  };
  // edit = ev => {
  //   let editedTask = {
  //     ...this.props.data,
  //     text: this.state.newText,
  //   };
  //   this.props.updateTask(editedTask);
  //   ev.target.blur();
  // };
  markAsCompleted = () => {
    // let editedTask = {
    //   ...this.props.data,
    //   completed: !this.props.data.completed,
    // };
    // this.props.updateTask(editedTask);
    this.props.complete(this.props.data.id );


  };
  delete = () => {
    this.props.delete(  this.props.data.id  )

  }
  render() {
    return (
      <div className={'task ' + (this.props.data.completed ? 'completed' : '')}>
        {/* <input type='text' value={this.state.newText} onChange={this.handleChange} /> */}

        <div className="text">{this.props.data.text}</div>

        <div className='actions '>
          <button className="delete" onClick={this.delete}>delete</button>
          <button className="complete" onClick={this.markAsCompleted}>completed</button>
        </div>
      </div>
    );
  }
}


const TaskConnected = connect(
  null,
  dispatch => ({
    delete: id => dispatch({ type: 'DELETE_TASK', id }),
    complete: id => dispatch({ type: 'TOGGLE_COMPLETED_TASK', id }),
  }),
)(Task);


export default TaskConnected;