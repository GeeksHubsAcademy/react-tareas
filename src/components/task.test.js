import React from 'react';
import { Task } from './task';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Task data={{ text: 'aprender TDD' }} dispatch={console.log} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('snapshot match', () => {
  test('it should keep consistent', () => {
    const text = 'aprender TDD';
    const wrapper = shallow(<Task data={{ text }} />);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('dinamic tests', () => {
  test('it should show the text in .text', () => {
    const text = 'aprender TDD';
    const wrapper = shallow(<Task data={{ text }} />);
    expect(wrapper.find('.text').text()).toBe(text);
  });
  test('it should have completed class', () => {
    const wrapper = shallow(<Task data={{ completed: true }} />);
    expect(wrapper.find('.completed').length).toBe(1);
  });
  test("it shouldn't have completed class", () => {
    const wrapper = shallow(<Task data={{ completed: false }} />);
    expect(wrapper.find('.completed').length).toBe(0);
  });
  test('it should run prop delete', () => {
    const id = 99;
    const onDeleted = jest.fn();
    const onCompleted = jest.fn();
    const wrapper = shallow(<Task data={{ text: 'hola', id }} delete={onDeleted} complete={onCompleted} />);
    wrapper.find('.delete').simulate('click');
    expect(onDeleted).toBeCalledWith(id);
    expect(onCompleted).not.toBeCalledWith(id);
  });
  test('it should run prop complete', () => {
    const id = 99;
    const onDeleted = jest.fn();
    const onCompleted = jest.fn();
    const wrapper = shallow(<Task data={{ text: 'hola', id }} delete={onDeleted} complete={onCompleted} />);
    wrapper.find('.complete').simulate('click');
    expect(onCompleted).toBeCalledWith(id);
    expect(onDeleted).not.toBeCalledWith(id);
  });
});
