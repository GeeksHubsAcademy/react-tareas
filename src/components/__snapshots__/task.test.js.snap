// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot match it should keep consistent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Task
    data={
      Object {
        "text": "aprender TDD",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="text"
        >
          aprender TDD
        </div>,
        <div
          className="actions "
        >
          <button
            className="delete"
            onClick={[Function]}
          >
            delete
          </button>
          <button
            className="complete"
            onClick={[Function]}
          >
            completed
          </button>
        </div>,
      ],
      "className": "task ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "aprender TDD",
          "className": "text",
        },
        "ref": null,
        "rendered": "aprender TDD",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="delete"
              onClick={[Function]}
            >
              delete
            </button>,
            <button
              className="complete"
              onClick={[Function]}
            >
              completed
            </button>,
          ],
          "className": "actions ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "delete",
              "className": "delete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "delete",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "completed",
              "className": "complete",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "completed",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="text"
          >
            aprender TDD
          </div>,
          <div
            className="actions "
          >
            <button
              className="delete"
              onClick={[Function]}
            >
              delete
            </button>
            <button
              className="complete"
              onClick={[Function]}
            >
              completed
            </button>
          </div>,
        ],
        "className": "task ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "aprender TDD",
            "className": "text",
          },
          "ref": null,
          "rendered": "aprender TDD",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="delete"
                onClick={[Function]}
              >
                delete
              </button>,
              <button
                className="complete"
                onClick={[Function]}
              >
                completed
              </button>,
            ],
            "className": "actions ",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "delete",
                "className": "delete",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "delete",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "completed",
                "className": "complete",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "completed",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
